### PR TITLE
DOC: improved documentation for `stats.ppcc_max`

### LINKED ISSUE #678: Statistics Review: ppcc_max (Trac #151)
_Original ticket http://projects.scipy.org/scipy/ticket/151 on 2006-04-02 by @rkern, assigned to @rkern._

The function `ppcc_max` in file source:trunk/Lib/stats/morestats.py needs review.

Please look over the StatisticsReview guidelines and add your comments below.

### PR DESCRIPTION
**Pull request by aeklant**
closes gh-678

### CODE REVIEW

**Comment by aeklant** [comment]
I added a basic test for regression and a lot of documentation since it was almost non-existent. 

I ran a few tests but I didn't understand well enough the concept of bracketing to come up with meaningful tests. Advice welcome.

**Comment by rgommers** [comment]
The added test was failing on Python 2.6 only:

    ======================================================================
    FAIL: test_morestats.test_ppcc_max_bad_arg
    ----------------------------------------------------------------------
    Traceback (most recent call last):
      File "/home/travis/virtualenv/python2.6.9/lib/python2.6/site-packages/nose/case.py", line 197, in runTest
        self.test(*self.arg)
      File "/home/travis/build/scipy/scipy/build/testenv/lib/python2.6/site-packages/scipy/stats/tests/test_morestats.py", line 747, in test_ppcc_max_bad_arg
        assert_almost_equal(stats.ppcc_max(x), -0.71215366521264145, decimal=16)
      File "/home/travis/virtualenv/python2.6.9/lib/python2.6/site-packages/numpy/testing/utils.py", line 468, in assert_almost_equal
        raise AssertionError(msg)
    AssertionError: 
    Arrays are not almost equal to 16 decimals
     ACTUAL: -0.71215367553074349
     DESIRED: -0.71215366521264145

I'm not sure why on Python 2.6 the behavior is different, but might be good to add a comment that for Python >=2.7 the test is OK up to the last significant digit.

**Comment by rgommers** [comment]
You can test the ``dist`` keyword with a similar approach as is done for ``ppcc_plot``. It checks that both a string and a distribution instance are accepted and give the same result.

**Comment by rgommers** [comment]
For testing the ``brack`` keyword, you can use a simple analytical function for which you know the minimum (say ``0.5*x**2 + 5``). Then you can test with the default interval (should not find the minimum, but stop at interval end) and with a large interval that includes the minimum (5).

**Comment by rgommers** [comment]
You documented that ``a<b<c`` for ``brack``, so can you verify that you get an understandable error message if that's not the case?

**Comment by rgommers** [comment]
Docs look good at first sight.

File: scipy/stats/morestats.py [hunk 1]
```diff
@@ -481,11 +481,73 @@ def probplot(x, sparams=(), dist='norm', fit=True, plot=None):
...
+
+    Examples
+    --------
+    First we generate some random data from a Tukey-Lambda distribution,
+    with shape parameter -0.7:
+
+    >>> from scipy import stats
+    >>> import matplotlib.pyplot as plt
+    >>> np.random.seed(1234567)
+    >>> x = stats.tukeylambda.rvs(-0.7, loc=2, scale=0.5, size=10000) + 1e4
```
Comment (ev-br) [inline scipy/stats/morestats.py]: {begin nitpick}
It's four lines where only one is relevant.  
I'd move importing matplotlib to the next paragraph (just before `fig = plt.figure()`). Also maybe move the seed into the rvs call: `x = stats.tukeylabmda.rvs(..., random_state=1234567)`.  
{end nitpick}

File: scipy/stats/morestats.py [hunk 2]
```diff
@@ -481,11 +481,73 @@ def probplot(x, sparams=(), dist='norm', fit=True, plot=None):
...
+    We calculate the value where the shape should reach its maximum and a red
+    line is drawn there. The line should coincide with the highest point in the
+    ppcc_plot.
+
+    >>> max = stats.ppcc_max(x)
+    >>> ax.vlines(max, 0, 1, colors='r', label='Expected shape value')
+
+    >>> plt.show()
 
-    See also ppcc_plot
```
Comment (ev-br) [inline scipy/stats/morestats.py]: Keep it, I'd say (only change to a proper See Also section). Maybe also mention `probplot` and the `boxcox` family

Comment (aeklant) [inline scipy/stats/morestats.py]: I added probplot and boxcox.

Note: This had already been changed to a proper See Also section, hence it doesn't show as an outdated diff.

**Comment by aeklant** [comment]
I have added the commented changes, excepting the one for `brack` keyword test.

**Comment by rgommers** [comment]
All changes look good to me. It would still be useful to add the test for ``brack`` that I suggested in https://github.com/scipy/scipy/pull/4998#issuecomment-115990238

File: scipy/stats/morestats.py [hunk 3]
```diff
@@ -481,11 +481,73 @@ def probplot(x, sparams=(), dist='norm', fit=True, plot=None):
...
+    one-parameter family of distributions.
+
+    Parameters
+    ----------
+    x : array_like
+        Input array.
+    brack : tuple, optional
+        Triple (a,b,c) where (a<b<c). If bracket consists of two numbers (a, c)
+        then they are assumed to be a starting interval for a downhill bracket
+        search (see `scipy.optimize.bracket`).
```
Comment (rgommers) [inline scipy/stats/morestats.py]: This should be ``optimize.brent``

**Comment by aeklant** [comment]
Please check the addition of tests and the wording of Notes section in the docstring.

**Comment by rgommers** [comment]
Changes look fine to me. If TravisCI is happy then this can be merged I think.

**Comment by rgommers** [comment]
No need to add a new commit for the one blank line that the PEP8 check doesn't like, I'll fix that up when merging.

**Comment by rgommers** [comment]
Merged in 6756b3c, thanks @aeklant!

**Comment by rgommers** [comment]
Merged in 6756b3c, thanks @aeklant!

**Pull request closed without merging**